blog/2023/04/julia-1.9-highlights.md

```diff
@@ -0,0 +1,393 @@
++++
+mintoclevel = 2
+maxtoclevel = 3
+title = "Julia 1.9 Highlights"
+authors = "..."
+published = "17 April 2023"
+rss_pubdate = Date(2023, 04, 17)
+rss = """Highlights of the Julia 1.9 release."""
++++
+
+
+After 3 betas and 3 release candidates, Julia version 1.9 has finally(!) been released. We would like to thank all the contributors to this release and all the testers that helped with finding regressions and issues in the pre-releases. Without you, this release would not have been possible.
+
+The full list of changes can be found in the [NEWS file](https://github.com/JuliaLang/julia/blob/release-1.9/NEWS.md), but here we'll give a more in-depth overview of some of the release highlights.
+
+\toc
+
+## TODOs:
+
+- [ ] Authors for "Memory usage hint"
+- [ ] Authors for "Tasks and the interactive thread pool"
+- [ ] Stack function
+- [ ] Link to docs for Numbered REPL prompt
+
+## Caching of native code
+
+*Tim Holy, Jameson Nash, and Valentin Churavy*
+
+In Julia, *precompilation* involves compiling package code and saving the compiled output to disk as a "cache file" for each package. This process significantly reduces the time needed for compilation when using the package, as you only need to build it once and can reuse it multiple times.
+
+However, prior to Julia 1.9, only a portion of the compiled code could be saved: types, variables, and methods were saved, as well as the outputs of any type-inference for the argument types specifically `precompile`d by the package developers. Notably absent from cache files was the *native code*—the code that actually runs on your CPU. Although caching helped reduce time-to-first-execution (TTFX) latency, the need to regenerate native code in each session meant that many packages still suffered from long TTFX latencies.
+
+With the introduction of Julia 1.9, native code caching is now available, resulting in a significant improvement in TTFX latency and paving the way for future enhancements across the ecosystem. Package authors can now utilize `precompile` statements or workloads with [PrecompileTools](https://github.com/JuliaLang/PrecompileTools.jl) to cache important routines in advance. Users can also create custom local "Startup" packages that load dependencies and precompile workloads tailored to their daily work.
+
+This feature comes with some tradeoffs, such as an increase in precompilation time by 10%-50%. However, since this is a one-time cost, we believe the tradeoff is well worth it. Cache files have also become larger due to the storage of more data and the use of a different serialization format.
+
+The graph below illustrates the changes in time-to-load (TTL), TTFX, and cache file size starting with Julia 1.7 (prior to the precompilation improvements):
+
+
+![](https://i.imgur.com/kydvmQL.png)
```

Comment:
```suggestion
![](https://i.imgur.com/kydvmQL.png)
Methodology: a demonstration workload was designed for each package. That workload was placed in the [Startup package](https://julialang.github.io/PrecompileTools.jl/stable/#Tutorial:-local-%22Startup%22-packages) and precompiled; for benchmarking, we load the Startup package and run the same workload.
```

It seems to make some sense to document the methodology used to make these measurements, while also calling extra attention to the benefits of custom startup packages.